staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -0,0 +1,144 @@
+	prometheus.MustRegister(apiserver_requests_execution_seconds)
+}
+
+var (
+	apiserver_rejected_requests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_rejected_requests",
+			Help:      "Number of rejected requests by request-management system",
+		},
+		[]string{priorityLevel, "reason"},
+	)
+	apiserver_current_inqueue_requests = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_current_inqueue_requests",
+			Help:      "Number of requests currently pending in the queue by request-management system",
+		},
+		[]string{priorityLevel},
+	)
+	apiserver_request_queue_length = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_request_queue_length",
+			Help:      "Length of queue in the request-management system",
+			Buckets:   queueLengthBuckets,
+		},
+		[]string{priorityLevel},
+	)
+	apiserver_request_concurrency_limit = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_request_concurrency_limit",
+			Help:      "Limit of shared concurrency in the request-management system",
+		},
+		[]string{priorityLevel},
+	)
+	apiserver_current_executing_requests = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_current_executing_requests",
+			Help:      "Number of requests currently executing in the request-management system",
+		},
+		[]string{priorityLevel},
+	)
+	apiserver_requests_waiting_inqueue_seconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_requests_waiting_inqueue_seconds",
+			Help:      "Time of requests pending in the queue",
+			Buckets:   requestDurationSecondsBuckets,
+		},
+		[]string{priorityLevel},
+	)
+	apiserver_requests_execution_seconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_requests_execution_seconds",
+			Help:      "Time of requests executing in the request-management system",
+			Buckets:   requestDurationSecondsBuckets,
+		},
+		[]string{priorityLevel},
+	)
+)
+
+func UpdateFlowControlRequestsInQueue(priorityLevel string, inqueue int) {
+	apiserver_current_inqueue_requests.WithLabelValues(priorityLevel).Set(float64(inqueue))
+}
+
+func UpdateFlowControlRequestsExecuting(priorityLevel string, executing int) {
+	apiserver_current_executing_requests.WithLabelValues(priorityLevel).Set(float64(executing))
+}
+
+func UpdateSharedConcurrencyLimit(priorityLevel string, limit int) {
+	apiserver_request_concurrency_limit.WithLabelValues(priorityLevel).Set(float64(limit))
+}
+
+func RecordRejects(priorityLevel string, reason string) {
```

Comment:
The name `RecordRejects`, since it uses the plural "Rejects", suggests that its parameters describe all the rejects --- which is not true.  Perhaps a better name would be something like `RecordReject` or `AddReject` or `ObserveReject`.

Reply:
renamed to `AddReject`